internal/log/log.go

```diff
@@ -91,6 +91,14 @@ func (logger *Logger) DebugEnabled() bool {
 	return logger.IsDebug
 }
 
+func (logger *Logger) CopyWith(prefix string, isDebug bool) log_api.Logger {
```

Comment:
Yes, but the problem is it forces you to use `defaultOutputBuffer`, which is stdout.  Therefore, if we used that instead of `CopyWith` here, it would screw up the forthcoming unit tests, where we'll want to make assertions and a mocked log writer.

Reply:
Hmm I see what you mean though I'm a bit curious why not go with something like `NewForServiceWithOptions` to match the other patterns we have here

Reply:
That would be a totally viable alternative.  We'd have to make the backing writer accessible.  

I still prefer the semantics of the above, though, since it imposes less burden on the client code.  That is, in this particular case, it's one less thing the client code has to destructure and put back into the ctor.  Beyond that (and this is not too important here but is worth thinking aobut in general), the copy semantics make the clients totally protected from future changes.  What we're saying, by using the copy method, is: "Change this and that, and keep everything else the same," and that everything else could be future things which don't even exist yet.  If you think about it, these semantics are exactly what you want in this case.

Reply:
Eh I'm not 100% on this being needed but not a blocker for now.